python/sglang/srt/layers/communicator.py

```diff
@@ -397,8 +399,19 @@ def _gather_hidden_states_and_residual(
             if hidden_states.shape[0] != 0:
                 hidden_states = layernorm(hidden_states)
         else:
-            hidden_states = tensor_model_parallel_all_reduce(hidden_states)
-            hidden_states, residual = layernorm(hidden_states, residual)
+            if (
+                is_sm100_supported()
+                and is_flashinfer_available()
+                and hasattr(layernorm, "forward_with_allreduce_fusion")
+                and global_server_args_dict["enable_flashinfer_allreduce_fusion"]
+                and hidden_states.shape[0] <= 1024
```

Comment:
hidden_state.numel() * hidden_state.element_size() < THRESHOLD

Reply:
Make sense, I'll update it.

Reply:
I think this feature could also be applied to other models. However, since different models may have varying hidden_size values, directly checking `max_token_num` seems more general. Regarding the token number 1024: This value was estimated based on the workspace threshold in trt_llm, using ds-v3's hidden_size as a reference. With 1024 tokens, it only allocates an additional ~10MB buffer, so the memory overhead is minimal.